hotp/host/main.c

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright (c) 2017, Linaro Limited
+ * All rights reserved.
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <err.h>
+#include <stdio.h>
+#include <string.h>
+
+/* OP-TEE TEE client API (built by optee_client) */
+#include <tee_client_api.h>
+
+/* For the UUID (found in the TA's h-file(s)) */
+#include <hotp_ta.h>
+
+struct test_value {
+	size_t count;
+	uint32_t expected;
+};
+
+/*
+ * Test values coming from the RFC4226 specification.
+ */
+struct test_value rfc4226_test_values[] = {
+	{ 0, 755224 },
+	{ 1, 287082 },
+	{ 2, 359152 },
+	{ 3, 969429 },
+	{ 4, 338314 },
+	{ 5, 254676 },
+	{ 6, 287922 },
+	{ 7, 162583 },
+	{ 8, 399871 },
+	{ 9, 520489 }
+};
+
+int main(int argc, char *argv[])
+{
+	TEEC_Context ctx;
+	TEEC_Operation op = { 0 };
+	TEEC_Result res;
+	TEEC_Session sess;
+	TEEC_UUID uuid = TA_HOTP_UUID;
+
+	int i = 0;
```

Comment:
No need to initialize `i` here

Reply:
True, but in general it is considered good practice to initialize variables. From what I recall people generally agrees with that besides if it is a static variable (and it should be initialized to zero and nothing else).

Reply:
Needless initialization should be avoided since you otherwise start to wonder "what am I missing, why is this variable initialized?"
When it comes to zero initialize static variables, I thought the general consensus was the opposite!

Reply:
> Needless initialization should be avoided since you otherwise start to wonder "what am I missing, why is this variable initialized?"

I disagree, but I will update to make everyone happy.

> When it comes to zero initialize static variables, I thought the general consensus was the opposite!

What? Are you saying that we are supposed to initialize static variables that should have the initial value as zero? If that is the case then I've learned something new today.

Reply:
Sorry, I misread you're previous comment.

Reply:
We're following the Linux kernel coding style so the rule is: no unnecessary initialization. While this choice certainly has downsides [1], it is expected that there is a greater benefit in not artificially hiding paths to the compiler where a variable is used before having received the proper value.

[1] https://www.cc.gatech.edu/grads/k/klu38/publications/ubi-ndss17.pdf

Reply:
The last review commit removed it, so I guess we are good to go?

Reply:
Yes